Octokit/Http/SimpleJsonSerializer.cs

```diff
@@ -23,55 +23,28 @@ public T Deserialize<T>(string json)
 
         class GitHubSerializerStrategy : PocoJsonSerializerStrategy
         {
-            readonly List<string> _membersWhichShouldPublishNull
-                 = new List<string>();
+            readonly List<string> _membersWhichShouldPublishNull = new List<string>();
 
             protected override string MapClrMemberToJsonFieldName(MemberInfo member)
             {
-                var memberName = member.Name;
-                var paramAttr = member.GetCustomAttribute<ParameterAttribute>();
-
-                if (paramAttr != null && !string.IsNullOrEmpty(paramAttr.Key))
-                {
-                    memberName = paramAttr.Key;
-                }
-
-                return memberName.ToRubyCase();
+                return member.GetJsonFieldName();
             }
 
             internal override IDictionary<string, ReflectionUtils.GetDelegate> GetterValueFactory(Type type)
             {
-                var fullName = type.FullName + "-";
-
-                // sometimes Octokit needs to send a null with the payload so the user
-                // can unset the value of a property.
-                // This method uses the same checks as PocoJsonSerializerStrategy
-                // to identify the right fields and properties to serialize
-                // but it then filters on the presence of SerializeNullAttribute.
+                var propertiesAndFields = type.GetPropertiesAndFields().Where(p => p.CanSerialize).ToList();
 
-                foreach (var propertyInfo in ReflectionUtils.GetProperties(type))
+                foreach (var property in propertiesAndFields.Where(p => p.SerializeNull))
```

Comment:
You could probably :fire: this `foreach` loop and populate `_membersWhichShouldPublishNull` directly:

```
_membersWhichShouldPublishNull = propertiesAndFields.Where(p => p.SerializeNull)
    .Select(member => type.FullName + "-" + member.JsonFieldName)
    .ToList();
```

But this code is already much more simpler than what was there before :sparkles: